tap_stripe/__init__.py

```diff
@@ -414,8 +437,16 @@ def sync_stream(stream_name):
     # If there is a sub-stream and its selected, get its bookmark (or the start date if no bookmark)
     should_sync_sub_stream = sub_stream_name and Context.is_selected(sub_stream_name)
     if should_sync_sub_stream:
-        sub_stream_bookmark = singer.get_bookmark(Context.state, sub_stream_name, replication_key) \
-            or int(utils.strptime_to_utc(Context.config["start_date"]).timestamp())
+
+        # Invoices's replication key changed from `date` to `created` in latest API version.
+        # Invoice line Items write bookmark with Invoice's replication key but it changed to `created`
+        # so kept `date` in bookmarking as it has to respect bookmark of active connection too.
+        if sub_stream_name == "invoice_line_items":
+            sub_stream_bookmark = singer.get_bookmark(Context.state, sub_stream_name, 'date') \
+                or int(utils.strptime_to_utc(Context.config["start_date"]).timestamp())
+        else:
```

Comment:
In the comment it is mentioned as Created, but in the code it is start_date. Looks the similar code is replicated. Can we write a function which accept these as parameters and set the appropriate bookmark?

Reply:
Yes, I added `get_bookmark_for_stream()` function to retrive bookmark and similarly added `write_bookmark_for_stream()`  for writing bookmark.